### PR TITLE
Sqlite - add sqlc and migration

### LINKED ISSUE #187: [SQL repo] move to migrations and query files
https://github.com/ark-network/ark/pull/180#discussion_r1639039836

### PR DESCRIPTION
**Pull request by sekulicd**
This updates Sqlite repo impl by adding sqlc and migration files.

Check [commit msg](https://github.com/ark-network/ark/pull/217/commits/ec18b1201dd88e8ad4c92bfac8b473282c91e43b) for explanation related to sqlc.embed and why i needed to introduce [views](https://github.com/sekulicd/ark/blob/1023ba4af001ace2f0d15957e6fb05b046e2a638/server/internal/infrastructure/db/sqlite/migration/20240703120550_init.up.sql#L59) as a workaround in order to avoid manual serialisation of fields when selecting from multiple tables. 

This closes #187 

@tiero @altafan @louisinger

### CODE REVIEW

File: server/internal/infrastructure/db/sqlite/utils.go [hunk 1]
```diff
@@ -43,3 +45,41 @@ func extendArray[T any](arr []T, position int) []T {
...
+
+	querier := queries.New(db)
+
+	defer func() {
+		if p := recover(); p != nil {
+			rollbackErr := tx.Rollback()
+			if rollbackErr != nil {
+				err = fmt.Errorf("panic: %v, rollback error: %w", p, rollbackErr)
+			}
+			panic(p) // Re-throw after rollback
```
Comment (tiero) [inline server/internal/infrastructure/db/sqlite/utils.go:67]: are we sure we want to panic the program here?

Comment (sekulicd) [reply server/internal/infrastructure/db/sqlite/utils.go:67]: this is 'repanic', we want to propagate panic so it is handled elsewhere, this is for logging purpose

**Comment by tiero** [comment]
CI is failing, other than that looks good to me

**Comment by sekulicd** [comment]
> CI is failing, other than that looks good to me
> 

passed after rerun, seems like e2e test is unstable, maybe cause nigiri

**Pull request merged**